plugins/modules/azure_rm_adapplication.py

```diff
@@ -146,33 +146,104 @@
     optional_claims:
         description:
             - Declare the optional claims for the application.
-        type: list
-        elements: dict
+        type: complex
         suboptions:
-            name:
-                description:
-                    - The name of the optional claim.
-                type: str
-                required: True
-            source:
+            access_token:
                 description:
-                    - The source (directory object) of the claim.
-                    - There are predefined claims and user-defined claims from extension properties.
-                    - If the source value is null, the claim is a predefined optional claim.
-                    - If the source value is user, the value in the name property is the extension property from the user object.
-                type: str
-            essential:
+                    - The optional claims returned in the JWT access token
+                type: list
+                elements: dict
+                suboptions:
+                    name:
+                        description:
+                            - The name of the optional claim.
+                        type: str
+                        required: True
+                    source:
+                        description:
+                            - The source (directory object) of the claim.
+                            - There are predefined claims and user-defined claims from extension properties.
+                            - If the source value is null, the claim is a predefined optional claim.
+                            - If the source value is user, the value in the name property is the extension property 
+                            from the user object.
```

Comment:
This line wrap is not correct, for such an error, redo the modification. Thank you!
```suggestion
- If the source value is user, the value in the name property is the extension property from the user object.
```